lib/mutant/mutator/node/send.rb

```diff
@@ -232,11 +232,18 @@ def emit_argument_propagation
         def mutate_receiver
           return unless receiver
           emit_implicit_self
+          emit_explicit_self
           emit_receiver_mutations do |node|
             !n_nil?(node)
           end
         end
 
+        def emit_explicit_self
+          return if UNARY_METHOD_OPERATORS.include?(selector)
+
+          emit_receiver(N_SELF) unless n_nil?(receiver)
+        end
```

Comment:
@mbj I added this and [this](https://github.com/mbj/mutant/pull/1210/files#diff-46eac0c6f51ce46d1c40d448b59a3219d725f9a571d38c0566b27c3c5905a3c7R20) to keep the change in behavior a bit more narrow since it may be useful to change `foo.bar.baz` to `self.baz`.

That said, we could also just emit implicit self in more cases and not generate this most of the time if we wanted. There are only a few cases where the explicit self is necessary or preferred `self.foo = 1`, etc.

If we emitted the implicit self directly instead of explicit, that would probably be nice for cases where mutant emits `self.baz` and you have to run it again to see that actually you could have just done `baz`. I have a WIP commit that does something like this that I can show you (I may just push it to this branch).